ocaml-lsp-server/src/custom_requests/req_get_documentation.ml

```diff
@@ -0,0 +1,112 @@
+open Import
+module TextDocumentPositionParams = Lsp.Types.TextDocumentPositionParams
+
+let meth = "ocamllsp/getDocumentation"
+let capability = "handleGetDocumentation", `Bool true
+
+module GetDocClientCapabilities = struct
+  type _t = { contentFormat : MarkupKind.t list }
+
+  let _yojson_of_t { contentFormat } =
+    `Assoc
+      (`List (List.map ~f:(fun format -> MarkupKind.yojson_of_t format) contentFormat))
+  ;;
+end
+
+module GetDocParams = struct
+  type t =
+    { text_document : TextDocumentIdentifier.t
+    ; position : Position.t
+    ; identifier : string option
+    ; contentFormat : MarkupKind.t option
+    }
+
+  let t_of_yojson json =
+    let open Yojson.Safe.Util in
+    let textDocumentPosition = Lsp.Types.TextDocumentPositionParams.t_of_yojson json in
+    let identifier = json |> member "identifier" |> to_option to_string in
+    let contentFormat =
+      json |> member "contentFormat" |> to_option MarkupKind.t_of_yojson
+    in
+    { position = textDocumentPosition.position
+    ; text_document = textDocumentPosition.textDocument
+    ; identifier
+    ; contentFormat
+    }
+  ;;
+
+  let yojson_of_t { text_document; position; identifier; contentFormat } =
+    let identifier =
+      match identifier with
+      | Some ident -> [ "identifier", `String ident ]
+      | None -> []
+    in
+    let contentFormat =
+      match contentFormat with
+      | Some format -> [ "contentFormat", MarkupKind.yojson_of_t format ]
+      | None -> []
+    in
+    `Assoc
+      ((("textDocument", TextDocumentIdentifier.yojson_of_t text_document)
+        :: ("position", Position.yojson_of_t position)
+        :: identifier)
+       @ contentFormat)
+  ;;
+end
+
+module GetDoc = struct
+  type t = { doc : MarkupContent.t }
+
+  let yojson_of_t { doc } = `Assoc [ "doc", MarkupContent.yojson_of_t doc ]
+
+  let t_of_yojson json =
+    let open Yojson.Safe.Util in
+    let doc = json |> member "doc" |> MarkupContent.t_of_yojson in
+    { doc }
+  ;;
+
+  let create ~kind ~value = MarkupContent.create ~kind ~value
```

Comment:
I think we are missing something here: when  the requested markup content is markdown, we should translate the documentation syntax to markown in the same way that it is done for the on_hover using `Doc_to_md.translate`.

This is how it is done in [hover_req.ml](https://github.com/ocaml/ocaml-lsp/blob/master/ocaml-lsp-server/src/hover_req.ml#L230-L232)

```ocaml
Option.map doc ~f:(fun doc ->
  match Doc_to_md.translate doc with
  | Raw d -> d
  | Markdown d -> d)
```

Reply:
Thank you for catching that.